subxt/src/tx/tx_progress.rs

```diff
@@ -502,15 +447,120 @@ impl<T: Config, Evs: Decode> TransactionEvents<T, Evs> {
     ///
     /// This works in the same way that [`events::Events::find_first()`] does, with the
     /// exception that it ignores events not related to the submitted extrinsic.
-    pub fn find_first<Ev: crate::Event>(&self) -> Result<Option<Ev>, BasicError> {
+    pub fn find_first<Ev: StaticEvent>(&self) -> Result<Option<Ev>, Error> {
         self.find::<Ev>().next().transpose()
     }
 
     /// Find an event in those associated with this transaction. Returns true if it was found.
     ///
     /// This works in the same way that [`events::Events::has()`] does, with the
     /// exception that it ignores events not related to the submitted extrinsic.
-    pub fn has<Ev: crate::Event>(&self) -> Result<bool, BasicError> {
+    pub fn has<Ev: StaticEvent>(&self) -> Result<bool, Error> {
         Ok(self.find::<Ev>().next().transpose()?.is_some())
     }
 }
+
+// Attempt to decode a DispatchError, returning either the ModuleError it decodes to,
+// along with additional details on the error, or returning the raw bytes to work with
+// downstream if we couldn't decode it.
+fn decode_dispatch_error(metadata: &Metadata, bytes: &[u8]) -> DispatchError {
```

Comment:
I may need access to this, see: https://github.com/paritytech/cargo-contract/blob/master/src/cmd/extrinsics/mod.rs#L246.

Essentially I want to decode the `DispatchError` returned from an RPC call rather than this extrinsic.

Could we hang it off `Metadata`? Or `DispatchError`?

Reply:
Ooh yup, sounds very reasonable to me! I stuck it onto `DispatchError`